### ISSUE TITLE
Commands offered when value should be expected

### ISSUE THREAD

**Issue by daniel-b2c2** [spring-projects/spring-shell#789]
`@ShellOption(ValueProvider = SomeProvider.class) Integer bob`

when I do `--bob TAB` I expect the values from SomeProvider to be present, which they are, but bob has arity 1, so we should expect at least one argument.

`--bob 1` is correct

`--bob --help` is not valid so why is `--help` and various other options from the command presented? We expect a value, so only show autocompletion for the value.

Once the value is in place then show the commands again.

**Comment by daniel-b2c2** [comment]
any view on this from triage?

**Comment by C-Otto** [comment]
Duplicate of #511?
